packages/ubuntu_bootstrap/lib/pages/autoinstall/autoinstall_model.dart

```diff
@@ -0,0 +1,102 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:file/file.dart';
+import 'package:file/local.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:path/path.dart' as p;
+import 'package:safe_change_notifier/safe_change_notifier.dart';
+import 'package:stdlibc/stdlibc.dart';
+import 'package:subiquity_client/subiquity_client.dart';
+import 'package:ubuntu_bootstrap/ubuntu_bootstrap.dart';
+import 'package:ubuntu_logger/ubuntu_logger.dart';
+import 'package:yaml/yaml.dart';
+
+final _log = Logger('autoinstall_model');
+
+final autoinstallModelProvider = ChangeNotifierProvider(
```

Comment:
We should try to not introduce more `ChangeNotifierProvider`s since they are discouraged by Riverpod:
https://riverpod.dev/docs/providers/change_notifier_provider

But maybe we should just convert all of them at a later stage?

Reply:
I'm aware of that, but I'd rather stick with a reasonably consistent architecture for now. I'd really like to improve all of this at some point :)